tests/phpunit/includes/abstract-testcase.php

```diff
@@ -187,6 +187,12 @@ public function tear_down() {
 		remove_filter( 'wp_die_handler', array( $this, 'get_wp_die_handler' ) );
 		$this->_restore_hooks();
 		wp_set_current_user( 0 );
+
+		// Reset comments globals.
+		$comment_globals = array( 'comment_alt', 'comment_depth', 'comment_thread_alt' );
+		foreach ( $comment_globals as $global ) {
+			$GLOBALS[ $global ] = null;
+		}
```

Comment:
I suggest moving this above `$this->unregister_all_meta_keys();` so it is is the same block of code as the resetting of other globals. 

This will require a dev-note as there is a risk extenders are using the global and it will break their tests. For the 6.0 backport it might be worth doing this in the test classes tear_down so there is some warning of potential breakage.

Reply:
Done in this commit https://github.com/WordPress/wordpress-develop/pull/2701/commits/3fca538435aac230e7217060edbe761000a37ae8 👍